src/Draco.Compiler/Internal/Semantics/DataflowErrors.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Draco.Compiler.Api.Diagnostics;
+
+namespace Draco.Compiler.Internal.Semantics;
+
+/// <summary>
+/// Holds constants for dataflow resolution errors.
+/// </summary>
+internal static class DataflowErrors
+{
+    private static string Code(int index) => DiagnosticTemplate.CreateDiagnosticCode(DiagnosticCategory.Dataflow, index);
+
+    /// <summary>
+    /// A function does not return on all paths.
+    /// </summary>
+    public static readonly DiagnosticTemplate DoesNotReturn = DiagnosticTemplate.Create(
+        title: "does not return",
+        severity: DiagnosticSeverity.Error,
+        format: "the function {0} does not return on all paths",
+        code: Code(1));
+
+    /// <summary>
+    /// A variable is used before it's initialized.
+    /// </summary>
+    public static readonly DiagnosticTemplate VariableUsedBeforeInit = DiagnosticTemplate.Create(
+        title: "use of uninitialized variable",
+        severity: DiagnosticSeverity.Error,
+        format: "the variable {0} is used before initialized",
+        code: Code(2));
+
+    // TODO: Is this really a dataflow error?
+    /// <summary>
+    /// Illegal value on left side of assignment.
+    /// </summary>
+    public static readonly DiagnosticTemplate IllegalLValue = DiagnosticTemplate.Create(
+        title: "illegal lvaule",
+        severity: DiagnosticSeverity.Error,
+        format: "illegal value on the left side of assignment",
+        code: Code(3));
+
+    /// <summary>
+    /// Immutable variable must be initialized at declaration site.
+    /// </summary>
+    public static readonly DiagnosticTemplate ImmutableVariableMustBeInitialized = DiagnosticTemplate.Create(
+        title: "immutable variable must be initialized",
+        severity: DiagnosticSeverity.Error,
+        format: "the immutable variable {0} must be initialized",
```

Comment:
Is it possible to have a formatting for the variable name ?
Also, I think it would be nice if the diagnostics started with an upper case letter and ended with a ".".